### PR TITLE
Most Viewed PoC

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Adds a Most Viewed component

On the server, this will render some placeholder text

On the client, this will make an AJAX call to retrieve the "Most Popular" HTML and inject it into the component.

## Why?

This is a proof of concept of a component that makes an AJAX request for additional data on the client but not the server

### CODE REVIEW

File: src/components/MostViewed/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    html,
+                });
+            });
+    }
+
+    render() {
+        return (
+            <MostViewed>
+                <h1>Most Viewed</h1>
+                <div dangerouslySetInnerHTML={{ __html: this.state.html }} />
```
Comment (SiAdcock) [inline src/components/MostViewed/index.js]: This kills me. Probably the most insecure thing ever.

File: src/components/MostViewed/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import { Component } from 'preact';
+import styled from 'preact-emotion';
+
+const MostViewed = styled('div')({});
```
Comment (sndrs) [inline src/components/MostViewed/index.js]: can just be a div?

Comment (SiAdcock) [reply src/components/MostViewed/index.js]: For sure!

File: src/components/MostViewed/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import { Component } from 'preact';
+import styled from 'preact-emotion';
+
+const MostViewed = styled('div');
+
+MostViewed.displayName = 'MostViewed';
```
[the remark this answers is not in the document]
Comment (SiAdcock) [reply src/components/MostViewed/index.js]: Yeah! Somewhere around 3am last night this clicked in my head. I'll get on it

**Comment by sndrs** [comment]
not sure how WIP this still is, but any reason not to go with `src/components/MostViewed.js`?

**Comment by sndrs** [comment]
it seems like a good enough go? it def gets it working – we might want to abstract it out at some point (not sure) but it does the job?

**Comment by SiAdcock** [comment]
Right, thanks man. I'll tidy up styles and merge. 

Agree this is good enough for a simple component. It might get trickier when we make additional AJAX calls that need to be orchestrated more carefully.

**Comment by SiAdcock** [comment]
Still not beautiful...

picture 534

File: src/static/css/most-viewed.css [hunk 4]
```diff
@@ -0,0 +1,41 @@
+.headline-column--desktop {
```
Comment (sndrs) [inline src/static/css/most-viewed.css]: not sure if you want to go this route, but you can style the html (we really should be using json, as you say :) in emotion, e.g. this is how the article styles capi stuff https://github.com/guardian/guui/blob/master/src/pages/Article.js#L25-L44

Comment (SiAdcock) [reply src/static/css/most-viewed.css]: Oh great, that is so much nicer! Thanks

**Pull request merged**